docs/Hub.md

```diff
@@ -0,0 +1,210 @@
+specified by the `model_path` field of the manifest, but with file names disambiguated with model SHA256 Hashes. 
+
+This way, the model cache is human readable, can disambiguate between multiple versions of models,
+ and can re-use cached models across different hubs if they have the same name and hash. 
+
+## Architecture
+
+![ONNX Hub Architecture](images/onnx_hub_arch.svg)
+
+The ONNX Hub consists of two main components, the client and the server.
+ The client code currently is included in the `onnx` package and can be pointed at a
+  server in the form of a hosted `ONNX_HUB_MANIFEST.json` within a github repository 
+  such as [the one in the ONNX Model Zoo](https://github.com/onnx/models/blob/master/ONNX_HUB_MANIFEST.json).
+  This manifest file is a JSON document which lists all models and their metadata
+   and is designed to be programming language agnostic. An example of a well formed model manifest entry is as follows:
+   
+   ```json
+{
+    "model": "BERT-Squad",
+    "model_path": "text/machine_comprehension/bert-squad/model/bertsquad-8.onnx",
+    "onnx_version": "1.3",
+    "opset_version": 8,
+    "metadata": {
+        "model_sha": "cad65b9807a5e0393e4f84331f9a0c5c844d9cc736e39781a80f9c48ca39447c",
+        "model_bytes": 435882893,
+        "tags": ["text", "machine comprehension", "bert-squad"],
+        "io_ports": {
+            "inputs": [
+                {
+                    "name": "unique_ids_raw_output___9:0",
+                    "shape": ["unk__475"],
+                    "type": "tensor(int64)"
+                },
+                {
+                    "name": "segment_ids:0",
+                    "shape": ["unk__476", 256],
+                    "type": "tensor(int64)"
+                },
+                {
+                    "name": "input_mask:0",
+                    "shape": ["unk__477", 256],
+                    "type": "tensor(int64)"
+                },
+                {
+                    "name": "input_ids:0",
+                    "shape": ["unk__478", 256],
+                    "type": "tensor(int64)"
+                }
+            ],
+            "outputs": [
+                {
+                    "name": "unstack:1",
+                    "shape": ["unk__479", 256],
+                    "type": "tensor(float)"
+                },
+                {
+                    "name": "unstack:0",
+                    "shape": ["unk__480", 256],
+                    "type": "tensor(float)"
+                },
+                {
+                    "name": "unique_ids:0",
+                    "shape": ["unk__481"],
+                    "type": "tensor(int64)"
+                }
+            ]
+        },
+        "model_with_data_path": "text/machine_comprehension/bert-squad/model/bertsquad-8.tar.gz",
+        "model_with_data_sha": "c8c6c7e0ab9e1333b86e8415a9d990b2570f9374f80be1c1cb72f182d266f666",
+        "model_with_data_bytes": 403400046
+    }
+}
+```
+This most important fields are:
```

Comment:
nit:
```suggestion
These important fields are:
```
Perhaps we can mention that ONNX encourages users to use cache to prevent downloading frequently.